src/metabase/driver/sync.clj

```diff
@@ -0,0 +1,51 @@
+(ns metabase.driver.sync
+  "General functions and utilities for sync operations across multiple drivers."
+  (:require [clojure.string :as str])
+  (:import java.util.regex.Pattern))
+
+(defn- schema-pattern->re-pattern ^Pattern [schema-pattern]
+  (re-pattern (-> (str/replace schema-pattern #"(^|[^\\\\])\*" "$1.*")
+                  (str/replace #"(^|[^\\\\])," "$1|"))))
+
+(defn- schema-patterns->filter-fn*
+  [inclusion-patterns exclusion-patterns]
+  (let [inclusion-blank? (str/blank? inclusion-patterns)
+        exclusion-blank? (str/blank? exclusion-patterns)]
+    (cond
+      (and inclusion-blank? exclusion-blank?)
+      (constantly true)
+
+      (and (not inclusion-blank?) (not exclusion-blank?))
+      (throw (ex-info "Inclusion and exclusion patterns cannot both be specified"
+                      {::inclusion-patterns inclusion-patterns
+                       ::exclusion-patterns exclusion-patterns}))
+
+      true
+      (let [inclusion? exclusion-blank?
+            pattern    (schema-pattern->re-pattern (if inclusion? inclusion-patterns exclusion-patterns))]
+        (fn [s]
+          (let [m        (.matcher pattern s)
+                matches? (.matches m)]
+            (if inclusion? matches? (not matches?))))))))
+
+(def ^:private schema-patterns->filter-fn (memoize schema-patterns->filter-fn*))
+
+(defn include-schema?
+  "Returns true of the given `schema-name` should be included/synced, considering the given `inclusion-patterns` and
+  `exclusion-patterns`. Patterns are comma-separated, and can contain wildcard characters (`*`)."
+  {:added "0.42.0"}
+  [inclusion-patterns exclusion-patterns schema-name]
+  (let [filter-fn (schema-patterns->filter-fn inclusion-patterns exclusion-patterns)]
+    (filter-fn schema-name)))
+
+(defn db-details->schema-filter-patterns
+  "Given a `prop-nm` (which is expected to be a connection property of type `:schema-filters`), and a `database`
```

Comment:
is it ever not "schema-filters" right now? Is this open lookup in use now or just possible future use.

Reply:
No, it's always supposed to be of type `schema-filters`, which is the custom type I added in the previous BigQuery patch to perform this logic.  I just don't think it's needed to validate it at this point (since this function will only ever be called for that conn-prop type).  Does that make sense, or did I miss your point?